pkg/apis/infrastructure/v1alpha2/g8s_control_plane_types.go

```diff
@@ -0,0 +1,133 @@
+package v1alpha2
+
+import (
+	"fmt"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewG8sControlPlaneCRD() *apiextensionsv1beta1.CustomResourceDefinition {
+	return &apiextensionsv1beta1.CustomResourceDefinition{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiextensionsv1beta1.SchemeGroupVersion.String(),
+			Kind:       "CustomResourceDefinition",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("g8sControlPlanes.%s", group),
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   group,
+			Scope:   "Namespaced",
+			Version: version,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Kind:     "G8sControlPlane",
+				Plural:   "g8sControlPlanes",
+				Singular: "g8sControlPlane",
+			},
+			Subresources: &apiextensionsv1beta1.CustomResourceSubresources{
+				Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+			},
+		},
+	}
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// G8sControlPlane defines the ControlPlane (Master nodes) of a
+// Giant Swarm Tenant Cluster
+//
+//	apiVersion: apiextensions.k8s.io/v1beta1
+//	kind: CustomResourceDefinition
+//	metadata:
+//	  name: g8sControlPlanes.core.giantswarm.io
+//	  annotations:
+//		giantswarm.io/docs:
+//	spec:
+//	  group: infrastructure.giantswarm.io
+//	  scope: Namespaced
+//	  version: v1alpha1
+//	  names:
+//	    kind: G8sControlPlane
+//	    plural: g8sControlPlanes
+//	    singular: g8sControlPlane
+//    subresources:
+//      status: {}
+//
+type G8sControlPlane struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              G8sControlPlaneSpec   `json:"spec"`
+	Status            G8sControlPlaneStatus `json:"status"`
+}
+
+type G8sControlPlaneSpec struct {
+	// Replicas is the number replicas of the master node.
+	Replicas int `json:"replicas" yaml:"replicas"`
+	// InfrastructureRef is a required reference to provider-specific
+	// Infrastructure.
+	InfrastructureReference string `json:"infrastructureReference"`
```

Comment:
```suggestion
	InfrastructureReference corev1.ObjectReference `json:"infrastructureReference"`
```
See https://godoc.org/k8s.io/kubernetes/pkg/apis/core#ObjectReference